src/suggestion.rs

```diff
@@ -113,23 +325,81 @@ impl<'s> fmt::Display for Suggestion<'s> {
             ))
             .fmt(formatter)?;
 
-        writeln!(formatter, " {}", self.chunk.as_str())?;
-
         // underline the relevant part with ^^^^^
-        let marker_size = if self.span.end.line == self.span.start.line {
+
+        // @todo this needs some more thought once multiline comments pop up
+        let mut marker_size = if self.span.end.line == self.span.start.line {
             // column bounds are inclusive, so for a correct length we need to add + 1
             self.span.end.column.saturating_sub(self.span.start.column) + 1
         } else {
             self.chunk
-                .as_str()
-                .chars()
-                .count()
+                .len_in_chars()
                 .saturating_sub(self.span.start.column)
         };
 
+        let literal_span: Span = self.span.clone();
+        let marker_range_relative: Range = self.range.clone();
+
         // if the offset starts from 0, we still want to continue if the length
         // of the marker is at least length 1
-        let offset = self.range.start;
+        let mut offset = marker_range_relative.start;
+        let mut v = self
+            .chunk
+            .as_str()
+            .lines()
+            .enumerate()
+            .map(|(lineno, content)| (lineno + 1, content))
+            .skip_while(|(lineno, _)| &self.span.start.line < lineno)
+            .take_while(|(lineno, _)| &self.span.end.line >= lineno)
+            .map(|(_, content)| content)
+            .collect::<Vec<&'_ str>>();
+        let mut count: usize = 0;
+        let mut last_line = 0;
+        for (i, c) in self.chunk.as_str().chars().enumerate() {
+            if c == '\n' {
+                count = count + 1;
+                last_line = i;
+            }
+            if i >= self.range.end {
+                break;
+            }
+        }
+        let mut initial_sentence = 0;
+        let mut lines: usize = 0;
+        let (mut stripped_line, mut pos) = get_current_statement(&v.as_ref(), self.range.clone());
+        let chars_till_start_statement = v[0..pos].iter().fold(0, |sum, x| sum + x.chars().count());
+        let range_word: Range = Range {
+            start: self.range.start.saturating_sub(chars_till_start_statement),
+            end: self.range.end.saturating_sub(chars_till_start_statement),
+        };
+
+        let n = stripped_line.char_indices().count();
+
+        let terminal_size = get_terminal_size();
+
+        // the line being analysed can affect how the indentation is done
+        // this values is dynamically calculated for each line where the documentation
+        let padding_till_literal_start = indent + 2; // 2 extra spaces are considered for starting the literal already
```

Comment:
`_literal_` is a bit confusing here, since we already have `proc_macro2::Literal` and `TrimmedLiteral` floating around, and what we really print is the `excerpt` (if we agree on that, see above)

Reply:
fixed, I had to google this word but seems to be exactly what it is needed here. Thanks!